ecs-agent/go.mod

```diff
@@ -3,23 +3,41 @@ module github.com/aws/amazon-ecs-agent/ecs-agent
 go 1.19
 
 require (
+	github.com/aws/amazon-ecs-agent/agent v0.0.0
 	github.com/aws/aws-sdk-go v1.36.0
 	github.com/cihub/seelog v0.0.0-20170130134532-f561c5e57575
 	github.com/didip/tollbooth v4.0.2+incompatible
 	github.com/golang/mock v1.4.1
 	github.com/gorilla/mux v1.8.0
+	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.7.0
 	golang.org/x/tools v0.6.0
 )
 
 require (
+	github.com/containerd/cgroups v1.0.4 // indirect
+	github.com/containernetworking/cni v0.8.1 // indirect
+	github.com/coreos/go-systemd/v22 v22.3.2 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/docker/go-connections v0.4.0 // indirect
+	github.com/docker/go-units v0.4.0 // indirect
+	github.com/godbus/dbus/v5 v5.0.6 // indirect
+	github.com/gogo/protobuf v1.3.2 // indirect
+	github.com/google/go-cmp v0.5.9 // indirect
+	github.com/google/uuid v1.3.0 // indirect
+	github.com/hectane/go-acl v0.0.0-20190604041725-da78bae5fc95 // indirect
 	github.com/jmespath/go-jmespath v0.4.0 // indirect
+	github.com/nxadm/tail v1.4.8 // indirect
+	github.com/opencontainers/runtime-spec v1.0.3-0.20210326190908-1c3f411f0417 // indirect
 	github.com/patrickmn/go-cache v2.1.0+incompatible // indirect
+	github.com/pborman/uuid v1.2.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	golang.org/x/mod v0.8.0 // indirect
 	golang.org/x/net v0.8.0 // indirect
 	golang.org/x/sys v0.6.0 // indirect
+	golang.org/x/text v0.8.0 // indirect
 	golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace github.com/aws/amazon-ecs-agent/agent => ../agent
```

Comment:
Why do we need this dependency? `ecs-agent` depends on `agent`, having the reverse dependency seems odd.

Reply:
Discussed offline, aligned that we should continue to leverage our existing dependency on seelog in the ecs-agent module until a logging interface to be used in ecs-agent module has been added in a future PR. I have updated this PR's changes and description accordingly. Thanks again for calling out!

Reply:
Discussed offline, aligned that we should continue to leverage our existing dependency on seelog in the ecs-agent module until a logging interface to be used in ecs-agent module has been added in a future PR. I have rebased and updated this PR's changes and description accordingly. Thanks again for calling out!